src/heaps/binomial_heap.jl

```diff
@@ -0,0 +1,371 @@
+# Binomial Heap
+mutable struct MutableBinomialHeapNode{T} 
+    data::T
+    degree::Int
+    handle::Int
+    parent::Union{MutableBinomialHeapNode,Nothing}
+    child::Union{MutableBinomialHeapNode,Nothing}
+    sibling::Union{MutableBinomialHeapNode,Nothing}
+end
+MutableBinomialHeapNode(data::T, handle::Int) where T = MutableBinomialHeapNode{T}(data, 0, handle, nothing, nothing, nothing)
+#################################################
+#
+#   heap type and constructors
+#
+#################################################
+mutable struct MutableBinomialHeap{T} <: AbstractHeap{T}
+    rootList::Vector{Union{Nothing,MutableBinomialHeapNode{T}}}
+    nodecount::Int
+    cumm_nodecount::Int
+    MutableBinomialHeap{T}() where T = new{T}([], 0, 0)
+
+    function MutableBinomialHeap{T}(xs::AbstractVector{T}) where T
```

Comment:
Where-ever possible, a collection should have a constructor that accepts an interator,